src/createRoom.ts

```diff
@@ -191,20 +192,18 @@ export default async function createRoom(opts: IOpts): Promise<string | null> {
         }
 
         if (opts.joinRule === JoinRule.Restricted) {
-            if (SpaceStore.instance.restrictedJoinRuleSupport?.preferred) {
-                createOpts.room_version = SpaceStore.instance.restrictedJoinRuleSupport.preferred;
-
-                createOpts.initial_state.push({
-                    type: EventType.RoomJoinRules,
-                    content: {
-                        "join_rule": JoinRule.Restricted,
-                        "allow": [{
-                            "type": RestrictedAllowType.RoomMembership,
-                            "room_id": opts.parentSpace.roomId,
-                        }],
-                    },
-                });
-            }
+            createOpts.room_version = PreferredRoomVersions.RestrictedRooms;
```

Comment:
Surely we ought to be checking `IRoomVersionsCapability::available` here otherwise we'll be asking for the server to create a room of a version it doesn't support and exploding

@turt2live this is a release blocker IMO

Reply:
https://github.com/vector-im/element-web/issues/22346

Reply:
from oob context: we could check `available` (and probably should), but realistically v9 has been out for so long that users shouldn't run into issues. We should also make the client extremely clear in what it needs from the server, per https://github.com/vector-im/element-web/issues/22346